tests/engine/training/test_graph_trainer.py

```diff
@@ -494,6 +494,112 @@ def test_graph_trainer_train_logging_with_cached_components(
         }
 
 
+def test_resources_fingerprints_are_unique_when_cached(
+    temp_cache: LocalTrainingCache, train_with_schema: Callable,
+):
+    train_schema = GraphSchema(
+        {
+            "train": SchemaNode(
+                needs={},
+                uses=PersistableTestComponent,
+                fn="train",
+                constructor_name="create",
+                config={"test_value": "4"},
+                is_target=True,
+            ),
+            "process": SchemaNode(
+                needs={"resource": "train"},
+                uses=PersistableTestComponent,
+                fn="run_inference",
+                constructor_name="load",
+                config={},
+            ),
+            "assert_node": SchemaNode(
+                needs={"i": "process"},
+                uses=AssertComponent,
```

Comment:
I think the issue is that this one returns a `CacheableText`. If this component returns something non cacheable then the test works

Reply:
good catch